### ISSUE TITLE
Remove Extension host terminates unexpectedly with Nx extension 18.11+

### ISSUE THREAD

**Issue by TheTiGuR** [nrwl/nx-console#1978]
When using the current (18.11, 18.11.1, 18.12) versions of the Nx Console in VS Code, this error pops up: 
![image](https://github.com/nrwl/nx-console/assets/542067/9c4ee8be-9604-4ceb-b803-8157e47b65eb)

The remote extension host then fails to load any other extensions

With 18.10.1, this did not happen

## Steps to Reproduce
I don't know how reproducible it is, but my setup:
Macbook Pro M3
Sonoma 14.2.1
VS Code 1.85.1
Running in a Dev Container with ~20 extensions installed, including Nx Console
On startup of the devcontainer with the latest Nx Console extension version installed, the error in the screenshot above will happen. If I downgrade to 18.10.1, it no longer happens. 

### Failure Logs / Images / Videos

Image above
Relevant snippet I think from the Extension Host Logs:
2024-01-12 10:54:14.747 [info] ExtensionService#_doActivateExtension nrwl.angular-console, startup: false, activationEvent: 'onStartupFinished'
2024-01-12 10:54:14.830 [info] ExtensionService#_doActivateExtension vscode.typescript-language-features, startup: false, activationEvent: 'api', root cause: nrwl.angular-console

I hope this is enough information, but if there is anything else I can provide, please let me know. Thanks!

**Comment by MaxKless** [comment]
Hey thanks for the report! Can you look into the developer console (`Help -> Toggle Developer Console`) and let me know if there are any errors in there?

**Comment by TheTiGuR** [comment]
I don't see anything special in there related to Nx, but I have include an old-version log to allow you to see the differences between current version of the console vs the old version (18.10.1) of the extension.

Current Extension Version (18.12.0): 
[vscode-app-1705336275639.log](https://github.com/nrwl/nx-console/files/13941126/vscode-app-1705336275639.log)

Old (working) Extension Version (18.10.1): 
[vscode-app-1705336427840-Old-version.log](https://github.com/nrwl/nx-console/files/13941159/vscode-app-1705336427840-Old-version.log)

If there is anything else I can provide, I will see what I can do. Thanks for looking into it!

**Comment by TheTiGuR** [comment]
To update, continuing to happen in 18.13.0 of the NX Console extension. Also just updated VS Code to latest, so now at 1.85.2 and continuing to happen.

**Comment by mmaday** [comment]
Just want to echo that my team and I are encountering the same issue with a very similar runtime environment.  We're also downgrading to 18.10.1 as a workaround.

**Comment by MaxKless** [comment]
Thanks for reporting this.
- Do you have a repro you can provide? 
- Is it happening across diferent nx versions?
- Could you both provide the results of your `nx report`?

Also if you could `Help -> Toggle Developer Console` and let me know any errors in there that would be super helpful

**Comment by TheTiGuR** [comment]
[devcontainer.json](https://github.com/nrwl/nx-console/files/13990790/devcontainer.json)

Here is an empty devcontainer with most of my customizations stripped out other than extensions, in case there is some kind of conflict between them. I continue to get the REH errors with this, zero code loaded/running, just created in an empty folder, (well, technically I put this file in a .devcontainer folder), started vscode, let it detect that it exists, and have it build the devcontainer based on that definition. And, when done building, REH errors.

**Comment by interrobrian** [comment]
My team is also getting this. We are also on Apple Silicon Macs. I did some experimentation with a minimal repro, and this appears to be caused by having both the Nx extension >=18.11 and the Prettier extension (`esbenp.prettier-vscode`) enabled in a dev container.

Repro steps are:
1. Open a folder in a container (I'm using `mcr.microsoft.com/devcontainers/typescript-node:1-20-bullseye`)
2. Install both Nx Console and Prettier
3. Reload the window

I've checked the remote extension logs and the vscode logs, but it's only showing an abrupt disconnect. I'm unable to find any specific cause. The actual contents of the workspace don't seem to matter - I can repro in a workspace with only the devcontainer.json.

**Comment by TheTiGuR** [comment]
We're up to 18.16 and this is still happening.

**Comment by MaxKless** [comment]
Hey! Sorry for not responding here. I will look into reproducing this and seeing how I can fix it.

**Comment by kylebjordahl** [comment]
Our team is hitting this same issue in a completely new devcontainer/repo we're setting up. 

One interesting finding, if I disable the `esbenp.prettier-vscode` extension on startup, I can get Nx console working fine _and then re-enable the prettier extension just fine_. 

Here's our nx report just for more data points: 
``` 
Node   : 20.11.1
OS     : linux-arm64
npm    : 10.2.4

nx                 : 18.0.7
@nx/js             : 18.0.7
@nx/jest           : 18.0.7
@nx/linter         : 18.0.7
@nx/eslint         : 18.0.7
@nx/workspace      : 18.0.7
@nx/devkit         : 18.0.7
@nx/eslint-plugin  : 18.0.7
@nx/plugin         : 18.0.7
@nrwl/tao          : 18.0.7
typescript         : 5.3.3
---------------------------------------
Community plugins:
@nxlv/python : 18.0.0
```

And our installed extension list (I had pared it all the way back at one point and narrowed it down to Nx+Prettier being the crash culprit)
```
amazonwebservices.aws-toolkit-vscode
apollographql.vscode-apollo
atlassian.atlascode
christian-kohler.npm-intellisense
christian-kohler.path-intellisense
dbaeumer.vscode-eslint
eamodio.gitlens
esbenp.prettier-vscode
firsttris.vscode-jest-runner
github.vscode-github-actions
github.vscode-pull-request-github
graphql.vscode-graphql
graphql.vscode-graphql-syntax
mgmcdermott.vscode-language-babel
ms-python.autopep8
ms-python.debugpy
ms-python.flake8
ms-python.isort
ms-python.python
ms-python.vscode-pylance
naumovs.color-highlight
nrwl.angular-console
redhat.vscode-yaml
stylelint.vscode-stylelint
tamasfe.even-better-toml
visualstudioexptteam.intellicode-api-usage-examples
visualstudioexptteam.vscodeintellicode
```

**Comment by MrChrisRodriguez** [comment]
@MaxKless any updates? This plugin completely breaks devcontainers which, in our case, includes CodeSpaces, Gitpod, Devpod, etc.

**Comment by TheTiGuR** [comment]
I see there was an update to the extension in the last couple of hours, and the other referenced ticket regarding this issue was marked completed, but initial testing this morning still has the same problems.

**Comment by MaxKless** [comment]
Yes it was closed because it's a duplicate of this one. I've set aside time this week to dive deeper into this issue and see what's going on :) 
Thank you for your patience

**Comment by TheTiGuR** [comment]
If more immediate feedback or testing can help, please ping me on discord, same name as here and I should be able to respond fairly quickly.

**Comment by ckaznocha** [comment]
I'm not able to create a repro but I suspect the issue is with the OS changing on nx. I was running into the same problem and also couldn't start the nx daemon inside the devcontainer.

Unfortunately I didn't make a copy of the log file but the nx daemon was failing with an error about not being able to find `@nx/nx-darwin-arm64`. My devcontainer is linux so something must have been setup from the host os (macOS).

From within the running devcontainer I deleted the `.nx` folder then repaired nx. The daemon now starts and I'm yet to get the error on the latest vscode extension.

```
nx --version
Nx Version:
- Local: v19.7.3
- Global: v19.8.3
```

vscode v1.93.1
Nx Console v18.28.1

**Comment by MaxKless** [comment]
I'll close this for now — the underlying issues here (wrong-platform native binaries in devcontainers, LSP client crash handling) have been fixed across several releases since this was reported. If you're still running into this on the latest nx console & nx, please let me know :)
